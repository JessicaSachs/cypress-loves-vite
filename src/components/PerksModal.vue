<template>
  <!-- fade in -->
  <transition name="fade">
    <div class="modal" v-show="visible">
      <div class="overlay"></div>
      <!-- <div class="overlay" key="overlay" /> -->
      <div class="content">
        <span class="close x-button" @click="$emit('close')">‚ùå</span>
        <div class="header-image" />
        <!-- This form should not trigger a full-page refresh! -->
        <form @submit.prevent="onSubmit">
          <p class="modal-title">A $10 off Perk for your first order</p>
          <p class="modal-subtitle">
            Get <b>$10 off</b> your first order of $15+. Now, that's tasty!
          </p>
          <fieldset>
            <label for="email">
              Email address
              <input
                id="email"
                type="email"
                v-model="formData.email"
                data-testid="email"
              />
            </label>

            <label for="zip-code">
              ZIP Code
              <input
                id="zip-code"
                type="number"
                v-model="formData.zipCode"
                data-testid="zip-code"
              />
            </label>

            <!-- When the user clicks the submit button, trigger a form submission -->
            <BaseButton
              color="primary"
              size="medium"
              type="submit"
              class="w-full"
              >Get $10 off</BaseButton
            >
          </fieldset>

          <div class="footer-links">
            <p @click.prevent="onClose">
              <a data-testid="dismiss-text" class="close dismiss-text" href="#"
                >Dismiss</a
              >
            </p>
            <p>
              <a data-testid="terms-link" class="terms" href="#"
                >Terms apply.</a
              >
            </p>
          </div>
        </form>
      </div>
    </div>
  </transition>
</template>

<style scoped>
.close {
  cursor: pointer;
}

.x-button {
  position: absolute;
  top: 0;
  right: 0;
  text-decoration: none;
}
</style>

<script>
import BaseButton from "./BaseButton.vue";

export default {
  props: {
    visible: { type: Boolean },
  },
  data() {
    return {
      formData: {},
    };
  },
  methods: {
    onClose() {},
    onSubmit() {
      this.$emit("save", this.formData);
      this.$emit("close");
    },
  },
  components: { BaseButton },
};
</script>

<style scoped>
.overlay {
  background: rgba(0, 0, 0, 0.2);
  width: 100vw;
  height: 100vh;
  position: absolute;
  top: 0;
  left: 0;
}

/** Input Fields */

input {
  @apply mb-2 mt-1 py-1 px-2 rounded-md border-2 border-true-gray-300 block;
}

/** transform */

/** Full-screen container */
.modal {
  @apply grid absolute place-content-center;
  top: 0;
  left: 0;

  height: 100vh;
  width: 100vw;
}

/** Inner container */
.content {
  @apply bg-white
    relative
    grid
    min-w-250px
    shadow-xl
    mb-80px;

  min-height: 80%;
  height: 100%;
  max-height: 800px;
  max-width: 500px;
  min-width: 400px;
}

/** Title and Subtitle */
.modal-title {
  @apply text-lg font-black pt-1;
}

.modal-subtitle {
  @apply text-sm pb-2 px-2 mb-4;
}

/**
 * Form inputs
 */
fieldset {
  @apply grid text-center gap-1 w-full;
}

label {
  @apply grid relative text-center font-bold text-sm;
}

form {
  @apply text-center grid place-items-center p-4;
}

/** Footer section */
.footer-links {
  @apply grid text-center place-content-center flex-row;
}

.close {
  @apply text-black font-black text-sm;
}

.dismiss-text {
  @apply underline;
}

.terms {
  @apply font-medium text-xs underline;
  color: blue;
}
</style>

<style scoped>
.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.5s ease;
}

.fade-enter-from,
.fade-leave-to {
  opacity: 0;
}
</style>

<style scoped>
.header-image {
  @apply min-h-80px;

  background-image: url(https://source.unsplash.com/400x225/?grubhub);
  background-size: cover;
}
</style>

<style>
body {
  @apply bg-gray-100;
}
</style>
