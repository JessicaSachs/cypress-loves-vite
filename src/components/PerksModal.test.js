import { mount } from '@vue/test-utils'
import PerksModal from './PerksModal.vue'

it('mounts', async () => {
  const wrapper = await mount(PerksModal, { props: { visible: true }})
  expect(wrapper.find('form').isVisible()).toBeTruthy()
})

it('mounts', async () => {
  const wrapper = await mount(PerksModal, { props: { visible: false }})
  expect(wrapper.find('form').isVisible()).toBeFalsy()
  expect(wrapper.element.innerHTML).toMatchSnapshot()
})





























// it('can fill out the form', async () => {
//   // const email = 'jessica@cypress.io'
//   // const zipCode = 10036
//   const wrapper = await mount(PerksModal)
//   // await wrapper.find('[data-testid=email]').setValue(email)
//   // await wrapper.find('[data-testid=zip-code]').setValue(zipCode)
//   // wrapper.find('form').trigger('submit')

//   // const emitted = wrapper.emitted('submit')
//   // expect(emitted).toHaveLength(2)
//   // expect(emitted[0][0].email).toBe(email)
//   // expect(emitted[0][0].zipCode).toBe("" + zipCode)

//   // expect(wrapper.element.innerHTML).toMatchSnapshot()
// })















//     // await wrapper.find('[data-testid=email]').setValue(email)
//   // await wrapper.find('[data-testid=zip-code]').setValue(zipCode)

//   // wrapper.trigger('submit')

//   // await wrapper.find('form').trigger('submit')

  
// // })




















// // const formData = {
// //   email: 'jessica@cypress.io',
// //   zipCode: 32084
// // }

// // const fillForm = async (options, component = PerksModal) => {
// //   const { email, zipCode } = options
// //   const wrapper = await mount(component, { props: { visible: true }})
// //   await wrapper.find('[data-testid=email]').trigger('input', email)
// //   await wrapper.find('[data-testid=zip-code]').trigger('input', zipCode)
// //   return wrapper
// // }

// // describe('PerksModal', () => {
// //   it('renders properly', async () => {
// //     const wrapper = await mount(PerksModal, { props: { visible: true }})
// //     const form = await wrapper.find('form')
// //     expect(form.isVisible()).toBeTruthy()
// //   })

// //   describe('valid submission', () => {
// //     it('can be filled out', async () => {
// //       const wrapper = await fillForm(formData)
// //       expect(wrapper.find('form').isVisible()).toBeTruthy()
// //     })

// //     it('doesnt navigate when the submit button is pressed', async () => {
// //       const spy = jest.fn()
// //       const wrapper = await fillForm(formData)
// //       const form = await wrapper.find('form')
// //       form.submit = spy
// //       await form.trigger('submit')
// //       expect(spy).not.toHaveBeenCalled()
// //     })

// //     it.only('emits the form data on submit', async () => {
// //       const wrapper = await fillForm(formData)
// //       const form = wrapper.find('form')
      

// //       await wrapper.vm.$nextTick()
// //       await button.trigger('click')


// //       const emitted = wrapper.emitted('submit')
// //       // debugger;
// //       // console.log(wrapper.emitted())
// //       // expect(emitted).toHaveLength(2)
// //       // expect(emitted[0][1]).toEqual(expect.objectContaining(formData))
// //     })
// //   })
// // })