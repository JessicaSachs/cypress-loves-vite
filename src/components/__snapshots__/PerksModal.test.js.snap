// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounts 1`] = `"<!-- fade in --><transition-stub><div class=\\"modal\\" style=\\"display: none;\\"><div class=\\"overlay\\"></div><!-- <div class=\\"overlay\\" key=\\"overlay\\" /> --><div class=\\"content\\"><span class=\\"close x-button\\">‚ùå</span><div class=\\"header-image\\"></div><!-- This form should not trigger a full-page refresh! --><form><p class=\\"modal-title\\">A $10 off Perk for your first order</p><p class=\\"modal-subtitle\\"> Get <b>$10 off</b> your first order of $15+. Now, that's tasty! </p><fieldset><label for=\\"email\\"> Email address <input id=\\"email\\" type=\\"email\\" data-testid=\\"email\\"></label><label for=\\"zip-code\\"> ZIP Code <input id=\\"zip-code\\" type=\\"number\\" data-testid=\\"zip-code\\"></label><!-- When the user clicks the submit button, trigger a form submission --><button class=\\"primary medium w-full\\" type=\\"submit\\">Get $10 off</button></fieldset><div class=\\"footer-links\\"><p><a data-testid=\\"dismiss-text\\" class=\\"close dismiss-text\\" href=\\"#\\">Dismiss</a></p><p><a data-testid=\\"terms-link\\" class=\\"terms\\" href=\\"#\\">Terms apply.</a></p></div></form></div></div></transition-stub>"`;
